### PR TITLE
K8SPSMDB-563 fix: add finalizer to delete pods in order

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPSMDB-563](https://jira.percona.com/browse/K8SPSMDB-563) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-563

Add `delete-psmdb-in-order` finalizer which deletes pods in order the way last pod will remain `PRIMARY`

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| monitoring-2-0 | failed |
| scheduled-backup | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| arbiter | passed |
| demand-backup-sharded | passed |
| service-per-pod | passed |
| limits | passed |
| liveness | passed |
| scaling | passed |
| smart-update | passed |
| security-context | passed |
| upgrade | passed |
| version-service | passed |
| rs-shard-migration | passed |
| upgrade-sharded | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| pitr-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/988664b7aff300a05a767ecdd4c1863d29651625
image: `perconalab/percona-server-mongodb-operator:PR-872-988664b7`

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 1]
```diff
@@ -541,6 +543,7 @@ func (r *ReconcilePerconaServerMongoDB) safeDownscale(cr *api.PerconaServerMongo
 		// downscale 1 pod on each reconciliation
 		if *sf.Spec.Replicas-rs.Size > 1 {
 			rs.Size = *sf.Spec.Replicas - 1
+			fmt.Println("TEST: rs", rs.Name, rs.Size)
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/psmdb_controller.go:546]: please remove debug

File: deploy/cr.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@ kind: PerconaServerMongoDB
 metadata:
   name: my-cluster-name
 #  finalizers:
+#    - delete-psmdb-in-order
```
Comment (delgod) [inline deploy/cr.yaml:6]: 1. please enable it by default
2. please rename finalizer
```suggestion
  finalizers:
    - delete-psmdb-pods-in-order
```

**Review by delgod — CHANGES_REQUESTED** [review]

File: deploy/cr.yaml (hunk 2 above)
Comment (pooknull) [reply deploy/cr.yaml]: fixed in d63633e678f9b40252449f0704914f4cde22b716

File: pkg/controller/perconaservermongodb/psmdb_controller.go (hunk 1 above)
Comment (pooknull) [reply pkg/controller/perconaservermongodb/psmdb_controller.go]: fixed in d63633e678f9b40252449f0704914f4cde22b716

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/d5870816f2375dba4886b4fe21e858667c5c209d
image: `perconalab/percona-server-mongodb-operator:PR-872-d5870816`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/storage.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/init-deploy.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/one-pod.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/upgrade-consistency.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/limits.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/monitoring-2-0.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/scaling.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/security-context.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/operator-self-healing-chaos.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/scheduled-backup.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/rs-shard-migration.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/arbiter.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/service-per-pod.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/demand-backup-sharded.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/users.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/upgrade.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/data-sharded.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/upgrade-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/non-voting.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/d5870816/pitr-sharded.log) |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/d5870816f2375dba4886b4fe21e858667c5c209d
image: `perconalab/percona-server-mongodb-operator:PR-872-d5870816`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/storage.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/one-pod.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/init-deploy.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/upgrade-consistency.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/limits.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/monitoring-2-0.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/scaling.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/security-context.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/operator-self-healing-chaos.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/scheduled-backup.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/rs-shard-migration.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/arbiter.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/service-per-pod.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/demand-backup-sharded.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/users.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/data-sharded.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/upgrade.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/non-voting.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/upgrade-sharded.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/pitr.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/a0591efd/pitr-sharded.log) |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/a0591efd866791de22c6e38e7bf788c3ad502f51
image: `perconalab/percona-server-mongodb-operator:PR-872-a0591efd`

**Review by delgod — DISMISSED** [review]

File: pkg/controller/perconaservermongodb/finalizers.go [hunk 3]
```diff
@@ -2,32 +2,77 @@ package perconaservermongodb
...
+func (r *ReconcilePerconaServerMongoDB) deletePSMDBPods(cr *api.PerconaServerMongoDB) (err error) {
+	done := true
+	for _, rs := range cr.Spec.Replsets {
+		sts, err := r.getRsStatefulset(cr, rs.Name)
+		if err != nil && !k8serrors.IsNotFound(err) {
+			return errors.Wrap(err, "get rs statefulset")
+		}
+		if k8serrors.IsNotFound(err) {
+			continue
+		}
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/finalizers.go:48]: how about?

```suggestion
		if err != nil {
			if k8serrors.IsNotFound(err) {
				continue
		        }
			return errors.Wrap(err, "get rs statefulset")
		}
```

File: pkg/controller/perconaservermongodb/finalizers.go [hunk 4]
```diff
@@ -2,32 +2,77 @@ package perconaservermongodb
...
 
-	return err
+		pods := &corev1.PodList{}
+		err = r.client.List(context.TODO(),
+			pods,
+			&client.ListOptions{
+				Namespace:     cr.Namespace,
+				LabelSelector: labels.SelectorFromSet(sts.Labels),
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/finalizers.go:55]: I think we should use `sts.Spec.Selector.MatchLabels`

File: pkg/controller/perconaservermongodb/finalizers.go [hunk 5]
```diff
@@ -2,32 +2,77 @@ package perconaservermongodb
...
+		}
+		if k8serrors.IsNotFound(err) {
+			continue
+		}
+		if len(pods.Items) > int(*sts.Spec.Replicas) {
+			return errors.New("waiting pods to be deleted")
+		}
+		if *sts.Spec.Replicas != 1 {
+			rs.Size = 1
+			done = false
```
Comment (egegunes) [inline pkg/controller/perconaservermongodb/finalizers.go:69]: If there are multiple replicasets, we may overwrite value of `done`

**Review by egegunes — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodb/finalizers.go (hunk 5 above)
Comment (pooknull) [reply pkg/controller/perconaservermongodb/finalizers.go:69]: That's correct. We cycle across every replicaset, and if one of them has size > 1, then we overwrite done to call `deletePSMDBPods` on the next reconcile loop

File: pkg/controller/perconaservermongodb/finalizers.go (hunk 4 above)
Comment (pooknull) [reply pkg/controller/perconaservermongodb/finalizers.go]: changed in 773ac15ac02d746002eccf93356e613d5987bf02

File: pkg/controller/perconaservermongodb/finalizers.go (hunk 3 above)
Comment (pooknull) [reply pkg/controller/perconaservermongodb/finalizers.go]: changed in 773ac15ac02d746002eccf93356e613d5987bf02

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/storage.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/init-deploy.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/one-pod.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/upgrade-consistency.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/limits.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/monitoring-2-0.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/scaling.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/security-context.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/operator-self-healing-chaos.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/rs-shard-migration.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/scheduled-backup.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/arbiter.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/service-per-pod.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/demand-backup-sharded.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/version-service.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/upgrade.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/users.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/pitr.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/data-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/non-voting.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/773ac15a/pitr-sharded.log) |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/773ac15ac02d746002eccf93356e613d5987bf02
image: `perconalab/percona-server-mongodb-operator:PR-872-773ac15a`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/init-deploy.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/one-pod.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/storage.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/upgrade-consistency.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/limits.log) |
| monitoring-2-0 | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/monitoring-2-0.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/scaling.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/security-context.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/operator-self-healing-chaos.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/rs-shard-migration.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/arbiter.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/service-per-pod.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/smart-update.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/scheduled-backup.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/version-service.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/users.log) |
| data-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/data-sharded.log) |
| demand-backup-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/demand-backup-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/non-voting.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-872/06121ac9/upgrade.log) |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/872/commits/06121ac94a7aebb622bead4e475caa5af6587782
image: `perconalab/percona-server-mongodb-operator:PR-872-06121ac9`

**Review by egegunes — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**